common/lib/types/presencemessage.js

```diff
@@ -70,13 +70,18 @@ var PresenceMessage = (function() {
 	PresenceMessage.encode = Message.encode;
 	PresenceMessage.decode = Message.decode;
 
-	PresenceMessage.fromResponseBody = function(body, options, format) {
+	PresenceMessage.fromResponseBody = function(body, options, format, channel) {
 		if(format)
 			body = (format == 'msgpack') ? msgpack.decode(body) : JSON.parse(String(body));
 
 		for(var i = 0; i < body.length; i++) {
 			var msg = body[i] = PresenceMessage.fromDecoded(body[i]);
-			PresenceMessage.decode(msg, options);
+			try {
+				PresenceMessage.decode(msg, options);
+			} catch (e) {
+				Logger.logAction(Logger.LOG_ERROR, 'PresenceMessage.fromResponseBody()', e.toString());
+				channel.emit('error', e);
```

Comment:
Only relevant one I can see is (RSL6b), which is for generic decoding in rest lib - "*If, for example, incompatible encryption details are provided or invalid Base64 is detected in the message payload, an error message will be sent to the logger, but the message will still be delivered with last successful decoding and the encoding field. For example, if a message had a decoding of “utf-8/cipher+aes-128-cbc/base64”, and the payload was successfully Base64 decoded but the payload could not be encrypted because the CipherParam details were not configured, the message would be delivered with a binary payload and an encoding with the value “utf-8/cipher+aes-128-cbc”*"

..which doesn't even mention events. (hence the question earlier about whether we should be emitting events on a rest channel at all. RTL7e (for realtime channels) mentions emitting an error event but doesn't say anything specific, just "*and an error should be emitted on the channel*")